tests/kafkatest/services/console_consumer.py

```diff
@@ -158,6 +165,14 @@ def _worker(self, idx, node):
         else:
             prop_file = self.render('console_consumer.properties')
 
+        # Add security properties to the config. If security protocol is not specified,
+        # use the default in the template properties.
+        security_config = SecurityConfig(node.account, self.security_protocol, prop_file)
+        prop_file += str(security_config)
+        if self.security_protocol is None:
+            self.security_protocol = security_config.security_protocol
+            if self.security_protocol == SecurityConfig.SSL:
+                self.new_consumer = True
```

Comment:
It's kind of weird to override the `new_consumer` setting here. It works, but it makes it hard to reason about what's going on since the setting changes while starting up and may contradict the user-specified setting.

I see how it can make sense to set the `new_consumer` flag if the test requests SSL, but if we might override the setting, then maybe the default value in the constructor should be `None` and we should only set it to `True` here if it was left as `None`. Otherwise, we could throw an exception indicating they specified invalid combinations of settings.

More generally, do we actually need to wait until this point to figure all this out? I see that `SecurityConfig` is using `node.account`, but can some of this logic be resolved in the constructor, which makes the entire class simpler to reason about since all settings will be computed upon construction?